docs/development/extensions-contrib/iceberg.md

```diff
@@ -0,0 +1,121 @@
+The `warehouseSource` type in the ingestion spec should be `hdfs`.
+
+If the Hive metastore supports Kerberos authentication, the following properties will be required in the `catalogProperties`:
+
+```json
+"catalogProperties": {
+  "principal": "krb_principal",
+  "keytab": "/path/to/keytab"
+}
+```
+Only Kerberos based authentication is supported as of now.
+
+#### Reading from S3 warehouse
+
+Ensure that the extension `druid-s3-extensions` is loaded. The data file paths are extracted from the Hive metastore catalog and `S3InputSource` is used to ingest these files.
+The `warehouseSource` type in the ingestion spec should be `s3`. If the S3 endpoint for the warehouse is different from the endpoint configured as the deep storage, the following properties are required in the `warehouseSource` section to define the S3 endpoint settings:
+
+```json
+"warehouseSource": {
+  "type": "s3",
+  "endpointConfig": {
+    "url": "S3_ENDPOINT_URL",
+    "signingRegion": "us-east-1"
+  },
+  "clientConfig": {
+    "protocol": "http",
+    "disableChunkedEncoding": true,
+    "enablePathStyleAccess": true,
+    "forceGlobalBucketAccessEnabled": false
+  },
+  "properties": {
+    "accessKeyId": {
+      "type": "default",
+      "password": "<ACCESS_KEY_ID"
+    },
+    "secretAccessKey": {
+      "type": "default",
+      "password": "<SECRET_ACCESS_KEY>"
+    }
+  }
+}
+```
+
+This extension uses the [Hadoop AWS module](https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/) to connect to S3 and retrieve the metadata and data file paths.
+The following properties will be required in the `catalogProperties`:
+
+```json
+"catalogProperties": {
+  "fs.s3a.access.key" : "S3_ACCESS_KEY",
+  "fs.s3a.secret.key" : "S3_SECRET_KEY",
+  "fs.s3a.endpoint" : "S3_API_ENDPOINT"
+}
+```
+Since the Hadoop AWS connector uses the `s3a` filesystem based client, the warehouse path should be specified with the `s3a://` protocol instead of `s3://`.
+
+### Local Catalog
+
+The local catalog type can be used for catalogs configured on the local filesystem. The `icebergCatalog` type should be set as `local`. This catalog is useful for demos or localized tests and is not recommended for production use cases.
+This catalog only supports reading from a local filesystem and so the `warehouseSource` is defined as `local`.
+
+### Known limitations
+
+This extension does not presently fully utilize the iceberg features such as snapshotting or schema evolution. Following are the current limitations of this extension:
+
+- The `IcebergInputSource` reads every single live file on the iceberg table up to the latest snapshot, which makes the table scan less performant. It is recommended to use iceberg filters on partition columns in the ingestion spec in order to limit the number of data files being retrieved. Since, Druid doesn't store the last ingested iceberg snapshot ID, it cannot identify the files created between that snapshot and the latest snapshot on iceberg.
```

Comment:
```suggestion
- The Iceberg input source reads every single live file on the Iceberg table up to the latest snapshot, which makes the table scan less performant. It is recommended to use Iceberg filters on partition columns in the ingestion spec to limit the number of data files being retrieved. Since Druid doesn't store the last ingested iceberg snapshot ID, it cannot identify the files created between that snapshot and the latest snapshot on Iceberg.
```